### PR TITLE
openlp: init at 2.4.6

### PR DESCRIPTION
**Pull request by jorsn**
free church presentation software

The package is split in two parts: The main python package to build (`lib.nix`) and a wrapper package pulling in configurable runtime dependencies (`default.nix`), so that rebuilds are unnecessary when changing runtime deps/features.

###### Review notes

1. The critical parts to test are: Video playback ([manual](https://manual.openlp.org/mediamanager.html#media)) via the 3 backends (system, webkit, vlc) and presentation support ([manual](https://manual.openlp.org/mediamanager.html#presentations)) via mupdf and libreoffice (uno).
2. All these tests require the package `openlpFull`. In `openlp` they aren't used, as they are optional.
3. When switching the video backend, it is best to restart OpenLP *afterwards* to make sure only the right one is initialized. To switch, select only one backend in *Settings → Configure OpenLP → Players*.

###### Things done

*(na)* means “not applicable”

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] *(na)* Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] *(na)* Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after) \
      package `openlp`: 676.1 MB, package `openlpFull`: 3.0 GB.
- [ ] *(na)* Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Additionally:
- [x] Tested in a pure nix-shell. There, audio does not work, as pulseaudio cannot be connected, but this is also the case using plain vlc.

### CODE REVIEW

**Comment by jorsn** [comment]
adapted qt 5.14

This PR now depends on #98185 as it is based on d63977883deadcddb4752526d87132134c0a368c.

File: pkgs/applications/misc/openlp/BUGS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+1. checks must be disabled because they are lacking the qt env.
+   (see ./lib.nix and https://discourse.nixos.org/t/qt-plugin-path-unset-in-test-phase/)
+1. segfault on exit. Not a real problem, according to debug log, everything
+   shuts down correctly. Maybe related to https://forums.openlp.org/discussion/3620/crash-on-exit.
+   Plan: Wait for OpenLP-3, since it is already in alpha 2
+   (2020-06-22; news: https://openlp.org/blog/)
+1. {my,postgre}sql and jenkins are disabled, because probably almost no one uses it and I
+   (jorsn) didn't test it. If someone needs support, please contact me.
```
Comment (jonringer) [inline pkgs/applications/misc/openlp/BUGS:8]: why is this relevant to nixpkgs?

File: pkgs/development/interpreters/python/default.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ pkgs, lib }:
+{ pkgs }:
 
 with pkgs;
```
Comment (jonringer) [inline pkgs/development/interpreters/python/default.nix:4]: unnecessary, please revert

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -10160,10 +10160,10 @@ in
 
   pythonInterpreters = callPackage ./../development/interpreters/python {
     # Overrides that apply to all Python interpreters
-    pkgs = pkgs // {
+    pkgs = pkgs.extend (pkgs: _: {
       qt5 = pkgs.qt514;
       libsForQt5 = pkgs.libsForQt514;
-    };
+    });
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:10166]: please revert

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -22836,18 +22836,10 @@ in
...
-          inherit libsForQt5 qt5;
-        };
-      };
-      self = python3;
-    };
+    pythonI = pkgs.pythonInterpreters.override { inherit pkgs; };
+    verString = lib.concatStrings (lib.take 2 (lib.splitVersion pkgs.python3.version));
   in libsForQt5.callPackage ../applications/networking/browsers/qutebrowser {
-    python3 = python;
+    python3 = pythonI."python${verString}";
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:22842]: what is this achieving? seems odd

File: pkgs/development/interpreters/python/default.nix (hunk 2 above)
Comment (jorsn) [reply pkgs/development/interpreters/python/default.nix:4]: Sorry, this is the reason why I wrote *”This PR now depends on #98185“* and removed the review request GitHub automatically dispatched. It is part of mentioned PR and I will rebase this PR once the dependency (now #98197) is merged.

What has it todo with openlp? Openlp uses pyqt_with_qtwebkit and the bug fixed in #98197 results in qt version 5.15 being used, for which qtwebkit is marked as broken.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (jorsn) [reply pkgs/top-level/all-packages.nix:10166]: same as above.

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (jorsn) [reply pkgs/top-level/all-packages.nix:22842]: same as above

File: pkgs/applications/misc/openlp/BUGS (hunk 1 above)
Comment (jorsn) [reply pkgs/applications/misc/openlp/BUGS:8]: They are issues with the packaging. As this is a relatively complex package, I thought it would be nice to write down them publicly instead of letting someone stumble over them and wonder. Also, reviewers are informed about me submitting this despite the issues. Normally I wouldn't submit anything which segfaults, but in this case I consider this a minor issue. At least they serve as a reminder for me try to resolve them (1 and 2) when I package the next release. Then, there is the link to the relevant discoure thread.

Do you think these notes are irrelevant to nixpkgs or should be just outsourced? Then they would be either a GH issues or private.

Comment (jonringer) [reply pkgs/applications/misc/openlp/BUGS:8]: I guess my concern is that people are likely to never become aware of this

File: pkgs/development/interpreters/python/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/interpreters/python/default.nix:4]: :(, idk. I guess you could help review other PR to help expedite it.

Comment (jorsn) [reply pkgs/development/interpreters/python/default.nix:4]: The missing bits there are only FRidh's review and (probably) acceptance of my last patch.

In the future, I will isolate my PRs.

File: pkgs/applications/misc/openlp/BUGS (hunk 1 above)
Comment (jorsn) [reply pkgs/applications/misc/openlp/BUGS:8]: If you like I can remove the note and keep it private. I have no great affections to this.

Comment (rnhmjoj) [reply pkgs/applications/misc/openlp/BUGS:8]: I agree with Jon, I don't think users will ever find this file. It's better to move this to a comment inside the main Nix expression, which can be more easily viewed with `nix edit nixpkgs.openlp`. Another option could be to put this inside a meta attribute: there isn't an appropriate one yet but I checked and Nix seems to allow arbitrary attributes:

```$ nix-env -f. -qa -A hello --json | jq '.hello.meta.bugs' --raw-output
This a list of bugs which affect this software packaged
in Nixpkgs:
  - blabla.
  - blablabla.
```

Comment (jorsn) [reply pkgs/applications/misc/openlp/BUGS]: Ok: I moved the bugs into a `meta.bugs` attr and moved meta into `lib.nix`, so that people using `nix edit` land directly in the source file.

File: pkgs/applications/misc/openlp/default.nix [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+}:
+
+assert pdfSupport -> (mupdf != null || ghostscript != null);
+assert presentationSupport -> libreoffice-unwrapped != null;
+assert enableVlc -> python3Packages.python-vlc or null != null;
+assert gstreamerSupport -> (gst_all_1 != null && gstPlugins gst_all_1 != null);
+
+let
+  # optional packages
+  pdfViewer = if isNull null then mupdf else ghostscript;
```
Comment (rnhmjoj) [inline pkgs/applications/misc/openlp/default.nix:30]: I think you mean
```suggestion
  pdfViewer = if isNull mupdf then mupdf else ghostscript;
```

Comment (jorsn) [reply pkgs/applications/misc/openlp/default.nix]: Thanks! Almost: if isNull mpdf then ghostscript… It's fixed and pushed.

File: pkgs/applications/misc/openlp/BUGS (hunk 1 above)
Comment (jorsn) [reply pkgs/applications/misc/openlp/BUGS]: ofborg complained about `meta.bugs`, so I moved it in `meta.longDescription`.

File: pkgs/applications/misc/openlp/default.nix (hunk 5 above)
Comment (rnhmjoj) [reply pkgs/applications/misc/openlp/default.nix]: Oh, yeah, right.

**Review by rnhmjoj — APPROVED** [review]
I don't see any other issue with the expression. I built the packages with `nixpkgs-review` and tested the application: to check the full functionality someone familiar with it should try it, but the program seem packaged and wrapped correctly.
To me it's all good for inclusion in Nixpkgs.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -25791,6 +25791,14 @@ in
 
   octopus = callPackage ../applications/science/chemistry/octopus { };
 
+  openlp = libsForQt514.callPackage ../applications/misc/openlp { };
+  openlpFull = openlp.override {
+    pdfSupport = true;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:25796]: Can we remove all options if they are on by default anyway? I think they don't get probably tested in all combinations and it would make the package a bit easier.

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -25791,6 +25791,14 @@ in
 
   octopus = callPackage ../applications/science/chemistry/octopus { };
 
+  openlp = libsForQt514.callPackage ../applications/misc/openlp { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:25794]: Can Qt5 be used here?

File: pkgs/applications/misc/openlp/lib.nix [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    rm -rf $out/${python.sitePackages}/tests
+    rm -rf $out/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Free church presentation software";
+    homepage = "https://openlp.org/";
+    downloadPage = "https://openlp.org/#downloads";
+    platforms = platforms.unix;
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openlp/lib.nix:74]: gpl2Plus or gpl2Only?

File: pkgs/applications/misc/openlp/lib.nix [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      done
+    )
+  '';
+
+  preFixup = ''
+    rm -rf $out/${python.sitePackages}/tests
+    rm -rf $out/bin
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openlp/lib.nix:69]:
```suggestion
  meta = with lib; {
```

stdenv.lib got recently deprecated.

File: pkgs/applications/misc/openlp/lib.nix [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+      cd ./resources/i18n
+      for file in *.ts; do
+          lconvert -i "$file" -o "$tdestdir/''${file%%ts}qm"
+      done
+    )
+  '';
+
+  preFixup = ''
+    rm -rf $out/${python.sitePackages}/tests
+    rm -rf $out/bin
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openlp/lib.nix:66]:
```suggestion
    rm -r $out/${python.sitePackages}/tests
    rm -r $out/bin
```

The build should fail if those files are missing.

File: pkgs/applications/misc/openlp/lib.nix [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+  #     running install tests
+  #     qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
+  #     This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
+  #
+  #     Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx.
+  #
+  # See also https://discourse.nixos.org/t/qt-plugin-path-unset-in-test-phase/
+
+  #checkInputs = [ mock nose ];
+  nativeBuildInputs = [ qt5.qttools ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openlp/lib.nix:34]:
```suggestion
  nativeBuildInputs = [ qttools ];
```

libsForQtX should not use qt5 as an input.

File: pkgs/applications/misc/openlp/default.nix [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+# for pdfSupport, presentationSupport, and media playback.
+{ stdenv, mkDerivation, wrapGAppsHook, python3Packages
+
+# qt deps
+, qtbase, qttools, qtmultimedia
+
+# optional deps
+, pdfSupport ? false, mupdf ? null, ghostscript ? null
+, presentationSupport ? false, libreoffice-unwrapped ? null, enableVlc ? false
+, gstreamerSupport ? false, gst_all_1 ? null, gstPlugins ? (gst: [
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openlp/default.nix:11]:
```suggestion
, pdfSupport ? false, mupdf, ghostscript
, presentationSupport ? false, libreoffice-unwrapped
, enableVlc ? false
, gstreamerSupport ? false, gst_all_1, gstPlugins ? (gst: [
```

``? null`` messes with overriding and can break it.

File: pkgs/applications/misc/openlp/default.nix [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+
+#, enableMySql ? false      # Untested. If interested, contact maintainer.
+#, enablePostgreSql ? false # Untested. If interested, contact maintainer.
+#, enableJenkinsApi ? false # Untested. If interested, contact maintainer.
+}:
+
+assert pdfSupport -> (mupdf != null || ghostscript != null);
+assert presentationSupport -> libreoffice-unwrapped != null;
+assert enableVlc -> python3Packages.python-vlc or null != null;
+assert gstreamerSupport -> (gst_all_1 != null && gstPlugins gst_all_1 != null);
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openlp/default.nix:26]:
```suggestion
```

None of those packages should be null at any time.

File: pkgs/applications/misc/openlp/default.nix [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+let
+  # optional packages
+  pdfViewer = if isNull mupdf then ghostscript else mupdf;
+  libreofficePath = "${libreoffice-unwrapped}/lib/libreoffice/program";
+
+  # lib functions
+  inherit (stdenv.lib.lists) optional optionals;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openlp/default.nix:34]:
```suggestion
  inherit (lib.lists) optional optionals;
```

File: pkgs/applications/misc/openlp/default.nix [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+}:
+
+assert pdfSupport -> (mupdf != null || ghostscript != null);
+assert presentationSupport -> libreoffice-unwrapped != null;
+assert enableVlc -> python3Packages.python-vlc or null != null;
+assert gstreamerSupport -> (gst_all_1 != null && gstPlugins gst_all_1 != null);
+
+let
+  # optional packages
+  pdfViewer = if isNull mupdf then ghostscript else mupdf;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openlp/default.nix:30]: mupdf can never be null except when overriden with null.

File: pkgs/top-level/all-packages.nix (hunk 6 above)
Comment (jorsn) [reply pkgs/top-level/all-packages.nix:25796]: In `openlp` as opposed to `openlpFull` they are turned off, and the difference in closure size is around 1.5 GB. Therefore it makes sense to me to have a minimal package and a full one. I ususally don't use the extra features, so for the basic use case (only music slides) the minimal package is sufficient.

I could unify the options such that they are jointly turned on and off. Would this be better for being simpler to test?
In principle they should be almost orthogonal (not interfering with each other), since they turn on probably independent runtime features by simply including the necessary dependencies in the closure. The only interference I know of is between `enableVlc` and `gstreamerSupport` in the discovery of different audio/video players, which can be selected at runtime. Of course I have no proof of non-interference of the rest.

I tested the feature sets of `openlp` and `openlpFull` but no other configurations.

File: pkgs/top-level/all-packages.nix (hunk 7 above)
Comment (jorsn) [reply pkgs/top-level/all-packages.nix:25794]: I will change it.

File: pkgs/applications/misc/openlp/lib.nix (hunk 11 above)
Comment (jorsn) [reply pkgs/applications/misc/openlp/lib.nix:34]: Why? This is in the python package and gets qt5 from libsForQt5, so it's synchronized. Though, I also include qttools as a dependency in the outer package without using it there.

I see mainly four main options:

1. Make everything one package. → rebuild necessary on every runtime dep change
2. Put `lib.nix` into the closure of the outer package in `default.nix`. → compromises clearly visible distinction between runtime and compile time stuff
3. Make `lib.nix` a function of `qttools`. → additional dependence between the two files – isn't this more complicated?
4. Leave as is.

File: pkgs/top-level/all-packages.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:25796]: 1.5GB is quite a bit. I think it is worth keeping it. Thanks for explaining.

File: pkgs/applications/misc/openlp/lib.nix (hunk 11 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/openlp/lib.nix:34]: This way you can mix qt versions.

> This is in the python package and gets qt5 from libsForQt5, so it's synchronized.

I am not fully sure this is the case. I would be more confident if you would use qttools directly.

Comment (jorsn) [reply pkgs/applications/misc/openlp/lib.nix:34]: I investigated: It is in principle synchronized since 2e9c639cb2a7ddf74956da2a945f8c9420f51297, but not in `pythonPackages.callPackage` because this is in another scope (see https://github.com/NixOS/nixpkgs/pull/102168#issuecomment-776090401).

But using `qt5` still doesn't add any problem: It is already there by using pyqt5. If pyqt5 (so pythonPackages) uses another qt version than the rest of the package you are totally screwed anyway. The only way to override qt consistently in a python+qt5 package is to override nixpkgs as described in the comment referenced above.

File: pkgs/top-level/all-packages.nix (hunk 7 above)
Comment (jorsn) [reply pkgs/top-level/all-packages.nix]: Resolved in 660a3abb0850535093bff14b1c8bca56a6ef8cb1.

File: pkgs/applications/misc/openlp/lib.nix (hunk 8 above)
Comment (jorsn) [reply pkgs/applications/misc/openlp/lib.nix]: Resolved in 660a3abb0850535093bff14b1c8bca56a6ef8cb1.

File: pkgs/applications/misc/openlp/lib.nix (hunk 9 above)
Comment (jorsn) [reply pkgs/applications/misc/openlp/lib.nix]: Resolved in 660a3abb0850535093bff14b1c8bca56a6ef8cb1.

File: pkgs/applications/misc/openlp/lib.nix (hunk 10 above)
Comment (jorsn) [reply pkgs/applications/misc/openlp/lib.nix]: Resolved in 660a3abb0850535093bff14b1c8bca56a6ef8cb1.

File: pkgs/applications/misc/openlp/default.nix (hunk 12 above)
Comment (jorsn) [reply pkgs/applications/misc/openlp/default.nix]: I removed ghostscript and all `null`s, and added a real check for a non-empty gstPlugin list in 660a3abb0850535093bff14b1c8bca56a6ef8cb1. I also renamed `enableVlc` to `vlcSupport` to be consistent.

File: pkgs/top-level/all-packages.nix [hunk 16]
```diff
@@ -27479,6 +27479,14 @@ in
 
   octopus = callPackage ../applications/science/chemistry/octopus { };
 
+  openlp = libsForQt5.callPackage ../applications/misc/openlp { };
+  openlpFull = openlp.override {
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix:27483]: different name attribute is needed as well for nix-env

Comment (jorsn) [reply pkgs/top-level/all-packages.nix]: I fixed it.

Comment (jorsn) [reply pkgs/top-level/all-packages.nix]: > different name attribute is needed as well for nix-env

...which applies maybe also for a couple of other packages:

`[ "connmanFull" "curlFull" "mjpegtoolsFull" "octaveFull" "pythonFull" "python2Full" "python27Full" "python3Full" "python36Full" "python37Full" "python38Full" "python39Full" "krb5Full" "pulseaudioFull" "samba4Full" "sambaFull" "gitFull" "polybarFull" ]`

says

```nix
{ pkgs ? import ./. {} }:
let
  attrNames = 
    [ "connman" "curl" "mjpegtools" "octave"
      "python" "python2" "python27" "python3" "python36" "python37" "python38" "python39"
      "harfbuzz" "krb5" "pulseaudio" "samba4" "samba" "codeblocks" "git" "polybar"
      "mercurial" # mercurial is the negative test case: its plain and Full names are distinct
    ];
  filtered = builtins.filter (n: pkgs."${n}".name == pkgs."${n}Full".name) attrNames;
in map (n: "${n}Full") filtered
```

For which of those is it unnecessary?

Comment (jorsn) [reply pkgs/top-level/all-packages.nix]: (I got the potential packages by `grep Full pkgs/top-level/all-packages.nix`)

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 91291` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>openlp</li>
<li>openlpFull</li>
</ul>
</details>

**Comment by jorsn** [comment]
In the last push (63ff41b2be8a6f6aa432e4e0bde257ea8ce38890) I added `meta.hydraPlatforms = [ ]` to the wrapper, because only the `baseLib` derivation needs to be built by Hydra.

**Comment by jorsn** [comment]
Anything preventing this from being merged?

**Comment by SuperSandro2000** [comment]
fails to build:

```
  Using setuptoolsBuildPhase
  Using setuptoolsShellHook
  Sourcing pip-install-hook
  Using pipInstallPhase
  Sourcing python-imports-check-hook.sh
  Using pythonImportsCheckPhase
  Sourcing python-namespaces-hook
  qtPreHook
  Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.
  builder for '/nix/store/cj3pjcndim4sp3b72f0i7jdqwdfy8ms8-python3.8-openlp-2.4.6.drv' failed with exit code
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 91291` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>openlp: log was empty</li>
<li>openlpFull: log was empty</li>
  </ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
openlpFull:
<details>


warning: unused-argument
Unused argument: `mock`.
Near pkgs/applications/misc/openlp/lib.nix:7:49:

```
  |
7 | , alembic, beautifulsoup4, chardet, lxml, Mako, mock, nose, pyenchant
  |                                                 ^
```
warning: unused-argument
Unused argument: `nose`.
Near pkgs/applications/misc/openlp/lib.nix:7:55:

```
  |
7 | , alembic, beautifulsoup4, chardet, lxml, Mako, mock, nose, pyenchant
  |                                                       ^
```
</details>
openlp:
<details>


warning: unused-argument
Unused argument: `mock`.
Near pkgs/applications/misc/openlp/lib.nix:7:49:

```
  |
7 | , alembic, beautifulsoup4, chardet, lxml, Mako, mock, nose, pyenchant
  |                                                 ^
```
warning: unused-argument
Unused argument: `nose`.
Near pkgs/applications/misc/openlp/lib.nix:7:55:

```
  |
7 | , alembic, beautifulsoup4, chardet, lxml, Mako, mock, nose, pyenchant
  |                                                       ^
```
</details>
</details>

**Comment by jorsn** [comment]
Fixed the failure due to the introduction of the new `qtPreHook`.

Result of `nixpkgs-review` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>openlp</li>
    <li>openlpFull</li>
  </ul>
</details>

**Comment by jorsn** [comment]
Still building and running:

Result of `nixpkgs-review pr 91291` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>openlp</li>
    <li>openlpFull</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 91291` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>openlp</li>
<li>openlpFull</li>
</ul>
</details>
